src/main/java/com/example/betandwin/weather/WeatherService.java

```diff
@@ -0,0 +1,64 @@
+package com.example.betandwin.weather;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.NoSuchElementException;
+
+@Service
+public class WeatherService {
+    private final WeatherDtoMapper weatherDtoMapper;
+
+    private static final String DEFAULT_CITY = "Wrocław";
+    private static final String BASE_URL = "http://api.openweathermap.org";
+    private static final String APP_ID = "&appid=ef778ae52fe2cc8bc09cf590fb99ad4e";
+
+    private static final String GEO_URL_BASE = "/geo/1.0/direct?q=";
+    private static final String GEO_URL_MIDDLE = "&limit=1";
+
+    private static final String WEATHER_URL_BASE = "/data/2.5/weather?";
+    private static final String WEATHER_URL_MIDDLE = "&units=metric&lang=pl";
+
+    public WeatherService(WeatherDtoMapper weatherDtoMapper) {
+        this.weatherDtoMapper = weatherDtoMapper;
+    }
+
+    private GeoLocationDto getCoordinatesByCity(String city, RestTemplate restTemplate) {
+        try {
+            String url = BASE_URL + GEO_URL_BASE + city + GEO_URL_MIDDLE + APP_ID;
+            ResponseEntity<GeoLocationDto[]> response = restTemplate.getForEntity(url, GeoLocationDto[].class);
+            GeoLocationDto[] dtoArray = response.getBody();
+            if (dtoArray.length > 0) {
+                return dtoArray[0];
+            }
+            return getCoordinatesByCity(DEFAULT_CITY, restTemplate);
```

Comment:
Nie do końca mi się to podoba. Metoda przyjmuje miasto, dla którego oczekujemy lokalizacji. Dwie opcje to:
- zwracanie Optionala - jeżeli miasto zostanie odnalezione, to będzie Optional z obiektem, a jeżeli nie, to pusty Optional
- rzucenie wyjątku, gdy nie będzie informacji o danym mieście